service/history/configs/config.go

```diff
@@ -204,19 +204,22 @@ type Config struct {
 	WorkflowTaskHeartbeatTimeout dynamicconfig.DurationPropertyFnWithNamespaceFilter
 
 	// The following is used by the new RPC replication stack
-	ReplicationTaskFetcherParallelism                  dynamicconfig.IntPropertyFn
-	ReplicationTaskFetcherAggregationInterval          dynamicconfig.DurationPropertyFn
-	ReplicationTaskFetcherTimerJitterCoefficient       dynamicconfig.FloatPropertyFn
-	ReplicationTaskFetcherErrorRetryWait               dynamicconfig.DurationPropertyFn
-	ReplicationTaskProcessorErrorRetryWait             dynamicconfig.DurationPropertyFnWithShardIDFilter
-	ReplicationTaskProcessorErrorRetryMaxAttempts      dynamicconfig.IntPropertyFnWithShardIDFilter
-	ReplicationTaskProcessorNoTaskRetryWait            dynamicconfig.DurationPropertyFnWithShardIDFilter
-	ReplicationTaskProcessorCleanupInterval            dynamicconfig.DurationPropertyFnWithShardIDFilter
-	ReplicationTaskProcessorCleanupJitterCoefficient   dynamicconfig.FloatPropertyFnWithShardIDFilter
-	ReplicationTaskProcessorStartWait                  dynamicconfig.DurationPropertyFnWithShardIDFilter
-	ReplicationTaskProcessorStartWaitJitterCoefficient dynamicconfig.FloatPropertyFnWithShardIDFilter
-	ReplicationTaskProcessorHostQPS                    dynamicconfig.FloatPropertyFn
-	ReplicationTaskProcessorShardQPS                   dynamicconfig.FloatPropertyFn
+	ReplicationTaskFetcherParallelism                    dynamicconfig.IntPropertyFn
+	ReplicationTaskFetcherAggregationInterval            dynamicconfig.DurationPropertyFn
+	ReplicationTaskFetcherTimerJitterCoefficient         dynamicconfig.FloatPropertyFn
+	ReplicationTaskFetcherErrorRetryWait                 dynamicconfig.DurationPropertyFn
+	ReplicationTaskProcessorErrorRetryWait               dynamicconfig.DurationPropertyFnWithShardIDFilter
+	ReplicationTaskProcessorErrorRetryBackoffCoefficient dynamicconfig.FloatPropertyFnWithShardIDFilter
```

Comment:
I'm surprised we have a shardIDFilter.  What is the use case for configuring this for one specific shard?

Reply:
the initial implementation comes from cadence repo.

i believe per shard configuration can provide some additional control, i.e. Cassandra (host restart, ring instability, etc) can see lots of error related to LTW. per shard configuration can provide some isolation